src/backend/looped/weapons/rapid_fire.cpp

```diff
@@ -10,11 +10,29 @@ namespace big
 	{
 		using looped_command::looped_command;
 
+		const std::size_t m_tick_rate = 5u;
+		std::size_t m_current_tick = 0;
+
+		/**
+		 * @brief We have to limit the fire rate for some shotguns as they can fill the bullet pool and crash the game.
+		 * 
+		 * @return true 
+		 * @return false 
+		 */
+		inline bool can_shoot()
```

Comment:
This could be optomized to:

```
inline bool can_shoot()
{
    bool result = (m_current_tick == 0);
    m_current_tick = (m_current_tick + 1) % m_tick_rate;
    return result;
}
```